### ISSUE TITLE
Review CanCreateAndGetAuthorizationWithoutFingerPrint test

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#1000]
This was originally written while transitioning to an environment where we may need to support fingerprint-less authorizations. That's now required, so perhaps we can :fire: this test.

TODO:

 - do we have another test around creating authorizations that means this is not necessary?
 - is there something special to this test that we need to keep around?

**Comment by M-Zuber** [comment]
looking into this

**Comment by M-Zuber** [comment]
What I have so far on this:
The `CanCreateAndGetAuthorizationByFingerprint` test does almost the exact same thing. *(some asserts maybe can be added based on the test w/o fingerprint)*

If the behavior is that `NewAuthorization` needs a fingerprint, then there is more involved then just removing the old test.
The model has to be also updated.

Will wait here on feedback before opening PR

**Comment by M-Zuber** [comment]
:wave:

**Comment by M-Zuber** [comment]
@shiftkey @tting you as I remembered to not rely on other notifications

**Comment by Anubhav10** [comment]
I was going through the API documentation, 'Get-or-create an authorization for a specific app' and 'Get-or-create an authorization for a specific app and fingerprint' are both supported. Why do you want to :fire: this test?
link to documentation- https://developer.github.com/v3/oauth_authorizations/

**Comment by shiftkey** [comment]
@Anubhav10 you're right, [creating an authorization without a fingerprint](https://developer.github.com/v3/oauth_authorizations/#get-or-create-an-authorization-for-a-specific-app) is still supported, so perhaps we need to keep this test around.

@M-Zuber apologies for missing those messages from back in December. What did you find when you were looking into those tests? And do you think we need to do anything in here given the previous statement?

**Comment by Anubhav10** [comment]
This is what I found. I was about to open a PR when I spotted the without fingerprint in the documentation. The code for CanCreateAndGetAuthorizationWithoutFingerprint and CanCreateAndGetAuthorizationByFingerprint is almost the same. In without fingerprint when creating authorization, it checked for tokenlasteight and hashedtoken values too. But it doesn't do so in  with fingerprint. By reading the comments it seems like it was assumed these were returned only when they were requested the second time(which is not so according to the documentation) and not when they were created.
Also I had removed the newauthorizations overloaded constructor without the fingerprint parameter(as now it was a required thing), but it was being used at multiple places, like creating personal tokens etc, so I kept it.

In the documentation, in the Deprecation Notice it says that "token is still supported" in both Get-or-create an authorization for a specific app' and 'Get-or-create an authorization for a specific app and fingerprint'. So does this mean that they'll remove this is future? Because if they do CanCreateAndGetAuthorizationByFingerprint  and CanCreateAndGetAuthorizationWithoutFingerprint both will fail as there is an assert statement to check that token value isn't whitespace or null
